device/tt_soc_descriptor.h

```diff
@@ -20,6 +20,7 @@
 #include "device/tt_arch_types.h"
 
 #include "device/coordinate_manager.h"
+#include "device/tt_core_coordinates.h"
```

Comment:
We will probably have this discussion with metal folks, but my guess is they'd request from us to enrich SoCDescriptor class with getters which return vectors or maps of CoreCoords

Reply:
I think users should at least know what coordinate system a SOC descriptor uses, and what core type each endpoint corresponds to. Today in TT-Metal, its restricted to Physical, but in future, we may need to introduce translated coordinate based SOC descriptors. Tracking coordinate systems in different layers of the stack may lead to repeated code and confusion.
Having said that, the direct user of UMD APIs in Metal is `tt_cluster`, which knows the SOC descriptor variant that gets loaded. This PR already introduces APIs that allow switching between coordinate systems. I think just having APIs from the coordinate manager should be good enough. We likely won't need to heavily augment the `tt_SocDescriptor` class.

Reply:
I was thinking about providing the API for getting the cores in any coordinate system from the same soc descriptor. It would be something like

```
tt_SocDescritpor soc_desc = tt_SocDescriptor(...);
std::vector<CoreCoord> cores = soc_desc.get_cores(CoreType::TENSIX, CoordSystem::LOGICAL);
```
But the soc descriptor would still be created from physical (noc coordinates)